src/test/scala/com/microsoft/hyperspace/index/HybridScanTest.scala

```diff
@@ -39,22 +39,34 @@ class HybridScanTest extends QueryTest with HyperspaceSuite {
   private val sampleDataLocationRoot = "src/test/resources/data/"
   private val sampleParquetDataLocationAppend = sampleDataLocationRoot + "sampleparquet0"
   private val sampleParquetDataLocationAppend2 = sampleDataLocationRoot + "sampleparquet1"
+  private val sampleParquetDataLocationDelete = sampleDataLocationRoot + "sampleparquet2"
+  private val sampleParquetDataLocationDelete2 = sampleDataLocationRoot + "sampleparquet3"
+  private val sampleParquetDataLocationDelete3 = sampleDataLocationRoot + "sampleparquet4"
+  private val sampleParquetDataLocationBoth = sampleDataLocationRoot + "sampleparquet5"
   private val sampleJsonDataLocationAppend = sampleDataLocationRoot + "samplejson1"
+  private val sampleJsonDataLocationDelete = sampleDataLocationRoot + "samplejson2"
   private var hyperspace: Hyperspace = _
 
   // Creates an index with given 'df' and 'indexConfig' and copies the first 'appendCnt'
   // number of input files from 'df'.
-  def setupIndexAndAppendData(df: DataFrame, indexConfig: IndexConfig, appendCnt: Int): Unit = {
+  def setupIndexAndChangeData(
+      df: DataFrame,
+      indexConfig: IndexConfig,
+      appendCnt: Int,
+      deleteCnt: Int): Unit = {
     hyperspace.createIndex(df, indexConfig)
     val inputFiles = df.inputFiles
-    assert(appendCnt < inputFiles.length)
+    assert(appendCnt + deleteCnt < inputFiles.length)
 
     val fs = systemPath.getFileSystem(new Configuration)
     for (i <- 0 until appendCnt) {
       val sourcePath = new Path(inputFiles(i))
       val destPath = new Path(inputFiles(i) + ".copy")
       fs.copyToLocalFile(sourcePath, destPath)
     }
+    for (i <- 1 to deleteCnt) {
```

Comment:
can we be consistent of using `until` or `to` (just check above)

Reply:
These are intended not to use `i - 1` or `deletedCnt +1`. Which do you think is better here?

Reply:
Ah OK, then it should be fine. Let's add the delete behavior to the `def` comment.